R/shiny.R

```diff
@@ -116,21 +116,26 @@ event_data <- function(
   
   event <- match.arg(event)
   eventID <- paste(event, source, sep = "-")
-  eventIDRegistered <- eventID %in% session$userData$plotlyShinyEventIDs
-  if (!eventIDRegistered) {
-    # If event_data() is requested before any (relevant) plotly
-    # graphs have rendered, we won't know if the relevant
-    # event-source combo is legitimate. As a result, we throw a 
-    # warning here to try to be useful in cases where you really
-    # are requesting an input value that is never registered, but
-    # without preventing valid code from running.
-    warning(
-      "The '", event, "' event tied a source ID of '", source, "' ",
-      "has not yet been registered. ", 
-      "If you encounter issues acquiring data from this event-source combo",
-      "Try adding `event_register(p, '", event, "')` to plot `p`"
-    )
-  }
+  
+  # It's possible for event_data() to execute before any 
+  # relevant input values have been registered (i.e, before 
+  # relevant plotly graphs have been executed). Therefore, 
+  # we delay checking that a relevant input value has been 
+  # registered until shiny flushes
+  session$onFlushed(
+    function() {
+      eventIDRegistered <- eventID %in% session$userData$plotlyShinyEventIDs
+      if (!eventIDRegistered) {
+        stop(
```

Comment:
I'd warn here, not stop. I'm not sure if throwing an error here interrupts other `onFlushed`, but even if not, I think it's closer to the intent if we just warn here, since there's no code on the stack that could even do anything sensible in response to this error, other than log it.